server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -425,35 +424,29 @@ DiskUsage averageUsage(RoutingNode node, ImmutableOpenMap<String, DiskUsage> usa
         return newUsage.getFreeDiskAsPercentage();
     }
 
+    private static final Decision YES_DISABLED = Decision.single(Decision.Type.YES, NAME, "the disk threshold decider is disabled");
+
+    private static final Decision YES_SINGLE_DATA_NODE =
+            Decision.single(Decision.Type.YES, NAME, "there is only a single data node present");
+
+    private static final Decision YES_USAGES_UNAVAILABLE = Decision.single(Decision.Type.YES, NAME, "disk usages are unavailable");
+
     private Decision earlyTerminate(RoutingAllocation allocation, ImmutableOpenMap<String, DiskUsage> usages) {
         // Always allow allocation if the decider is disabled
         if (diskThresholdSettings.isEnabled() == false) {
-            return allocation.decision(Decision.YES, NAME, "the disk threshold decider is disabled");
+            return YES_DISABLED;
         }
 
         // Allow allocation regardless if only a single data node is available
         if (enableForSingleDataNode == false && allocation.nodes().getDataNodes().size() <= 1) {
-            if (logger.isTraceEnabled()) {
-                logger.trace("only a single data node is present, allowing allocation");
-            }
-            return allocation.decision(Decision.YES, NAME, "there is only a single data node present");
-        }
-
-        // Fail open there is no info available
-        final ClusterInfo clusterInfo = allocation.clusterInfo();
-        if (clusterInfo == null) {
```

Comment:
This check is dead code, both calling methods to this method use the `clusterInfo` before it runs so it's never `null` here for sure.